src/v1alpha3/echo_client.ts

```diff
@@ -0,0 +1,233 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ** This file is automatically generated by gapic-generator-typescript. **
+// ** https://github.com/googleapis/gapic-generator-typescript **
+// ** All changes to this file may be overwritten. **
+
+import * as gax from 'google-gax';
+import * as path from 'path';
+import * as protobuf from 'protobufjs';
+
+import * as packageJson from '../../package.json';
+import * as protos from '../../pbjs-genfiles/protos';
+import * as gapicConfig from './echo_client_config.json';
+
+const version = packageJson.version;
+
+export interface ClientOptions extends gax.GrpcClientOptions,
+                                       gax.GoogleAuthOptions,
+                                       gax.ClientStubOptions {
+  libName?: string;
+  libVersion?: string;
+  clientConfig?: gax.ClientConfig;
+}
+
+interface Descriptors {
+  page: {[name: string]: gax.PageDescriptor};
+  stream: {[name: string]: gax.StreamDescriptor};
+  longrunning: {[name: string]: gax.LongrunningDescriptor};
+}
+
+export interface Callback<ResponseObject> {
```

Comment:
I would expect the `ResponseObject` type here to be generic

Reply:
I didn't quite understand you here - can you explain please? I was under impression it _is_ generic now, since it's `Callback<ResponseObject>`.

Reply:
Where does the `ResponseObject` type definition come from?

Reply:
I think `ResponseObject` is the type variable, so it shouldn't be defined anywhere

Reply:
@JustinBeckwith It's substituted with an actual type for each auto-generated method: 
```
  echo(
      request: protos.google.showcase.v1alpha3.IEchoRequest,
      options: gax.CallOptions,
      callback: Callback<protos.google.showcase.v1alpha3.IEchoResponse>): void;
```